### PR TITLE
chore: upgrade rust-toolchain to nightly-2022-09-18

### LINKED ISSUE #7466: Upgrade rust-toolchain to nightly-2022-09-02
rust-1.63 is released, and the new nightly(2022-09-02) introduces different APIs:
- `std::Error::backtrace()` is replaced with `provider` style API: `request_ref::<Backtrace>()`.
- `std::Simd` API changes that affect `arrow2`: [a draft update](https://github.com/datafuse-extras/arrow2/pull/6) is fired for illustration.

The changes about `Error` affect several third-party crates too, such as `thiserror`, in [1.0.33](https://github.com/dtolnay/thiserror/releases/tag/1.0.33) it switched to the provider style API to retrieve backtrace.

[openraft](https://github.com/datafuselabs/openraft/tree/v0.7.2) is updated and is ready to upgrade to nightly-2022-09-02.

Once these modifications are ready, it's time to upgrade to the new nightly 2022-09-02.

@sundy-li @Xuanwo @PsiACE 
Opinion? Suggestions? 🤔

### PR DESCRIPTION
**Pull request by xudong963**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

The ticket contains the following changes:
- arrow2 & parquet2 @sundy-li @dantengsky 
- `readbuf`: https://github.com/rust-lang/rust/pull/97015 @sundy-li 
- meta storage @drmingdrmer 
- some clippy fixes

Fixes #7466

### CODE REVIEW

**Comment by Xuanwo** [comment]
Please upgrade our build-tool first.

Ref: https://databend.rs/doc/faq/routine-maintenance

**Comment by xudong963** [comment]
@mergify update

**Review by Xuanwo — APPROVED** [review]
Nice!

**Review by sundy-li — APPROVED** [review]

**Comment by sundy-li** [comment]
GAT is stable now !

File: src/meta/types/src/lib.rs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+#![feature(provide_any)]
```
Comment (BohuTANG) [inline src/meta/types/src/lib.rs:1]: This FEATURE behind the license is wried :)

**Comment by BohuTANG** [comment]
nightly-2022-09-18 great 👍

File: src/meta/types/src/lib.rs (hunk 1 above)
Comment (xudong963) [reply src/meta/types/src/lib.rs]: +1

**Review by PsiACE — APPROVED** [review]

**Pull request merged**